core/primitives/src/types.rs

```diff
@@ -146,6 +146,11 @@ pub type MaybeBlockId = Option<BlockId>;
 
 #[derive(Default, BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
 pub struct ValidatorStats {
-    pub blocks_produced: BlockIndex,
-    pub blocks_expected: BlockIndex,
+    pub produced: BlockIndex,
+    pub expected: BlockIndex,
```

Comment:
Are these better described as `NumBlocks`? (ref #1882)

Since the PR is not merged yet, we should definitely keep track of this change and update the PR if this one is merged earlier.

P.S. I seem to like the old naming (`block_*`) better even though I seem to understand that the change is related to the fact that this structure is reused for `chunk_stats`. A question is ringing in my head: "produced what?". It is especially confusing in the usage contexts where you don't see the types of the fields.

Reply:
Yeah, it should be NumBlocks, but that PR is not ready indeed.
It just seems weird to create exactly the same structure which will contain `chunks_produced` but can be done.